Project2015To2017/Reading/ProjectReader.cs

```diff
@@ -150,13 +150,21 @@ private static void ProcessProjectReferences(Project projectDefinition)
 		{
 			foreach (var reference in projectDefinition.ProjectReferences)
 			{
-				reference.ProjectFile = new FileInfo(Path.Combine(projectDefinition.FilePath.Directory.FullName, reference.Include.Replace('\\', Path.DirectorySeparatorChar)));
+				if (reference.ProjectFile != null)
+				{
+					continue;
+				}
+
+				var path = reference.Include;
+				var projectDirectory = projectDefinition.FilePath.DirectoryName;
+				var adjustedPath = Extensions.MaybeAdjustFilePath(path, projectDirectory);
+				reference.ProjectFile = new FileInfo(Path.Combine(projectDirectory, adjustedPath));
 			}
 		}
 
 		private FileInfo FindPackagesConfigFile(FileInfo projectFile)
 		{
-			var packagesConfig = new FileInfo(Path.Combine(projectFile.Directory.FullName, "packages.config"));
+			var packagesConfig = new FileInfo(Path.Combine(projectFile.DirectoryName, "packages.config"));
```

Comment:
This doesn't work if your directory is nested, right? Fullname took the entire path, this makes the assumption of being one folder deep?

Reply:
I'm afraid it's incorrect. [DirectoryName reference](https://docs.microsoft.com/en-us/dotnet/api/system.io.fileinfo.directoryname?redirectedfrom=MSDN&view=netframework-4.7.2#System_IO_FileInfo_DirectoryName).
`Directory` property internally calls `DirectoryName`, checks `string.IsNullOrEmpty`, and creates `DirectoryInfo` instance from that string.

Reply:
Interesting, you’re right I was thinking of `DirectoryInfo.Name`